src/timezones/types.jl

```diff
@@ -111,17 +111,39 @@ doc"""A `TimeZone` with an offset that changes over time."""
 immutable VariableTimeZone <: TimeZone
     name::Symbol
     transitions::Vector{Transition}
+    max_year::Nullable{Int}
+end
+
+function VariableTimeZone(name::AbstractString, transitions::Vector{Transition}, max_year::Nullable{Int})
+    return VariableTimeZone(symbol(name), transitions, max_year)
 end
 
 function VariableTimeZone(name::AbstractString, transitions::Vector{Transition})
-    return VariableTimeZone(symbol(name), transitions)
+    return VariableTimeZone(symbol(name), transitions, Nullable())
 end
 
+type OutOfRangeTimeError <: TimeError
+    tz::VariableTimeZone
+end
+Base.showerror(io::IO, e::OutOfRangeTimeError) = print(io, "TimeZone $(string(e.tz)) does not handle dates after $(string(e.tz.max_year.value))");
```

Comment:
`$(string(e.tz.max_year.value))` should probably be changed to `$(string(get(e.tz.max_year, -1)))` or something similar. Otherwise entries where `e.tz.max_year == Nullable{Int}()` will be some random value.